### PR TITLE
Create Data Grid Component

### PR DESCRIPTION
**Pull request by paulagutierrez08**
## Description

Create component requirements design document.

The component will be called DataGrid.

Create [Sorting](https://mui.com/x/react-data-grid/sorting/) component: Allows users to sort column data in ascending or descending order.

Create [Filtering](https://mui.com/x/react-data-grid/filtering/) component: Allows users to filter the data displayed in the DataGrid based on certain criteria.

Create [Pagination](https://mui.com/x/react-data-grid/pagination/) component: It allows data to be divided into pages, facilitating navigation and improving performance when loading large data sets.

## Related Issue

Feat TELCO-53

## Type of Change

- [x] feat: A new feature

## Test Coverage

image

### Component
image

image

### Storybook
image

## Checklist:

- [x] My code follows the [UI Coding Guidelines](https://docs.google.com/document/d/1prF4Oh72WARlM1lU7I0uVwZ-ruSkdHuJ39HtGUmtEm8/edit?usp=sharing) of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes generate no javascript errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes and coverage is at least 80%
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/stories/atoms/data-grid.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import { Meta, StoryObj } from '@storybook/react';
+
+import { GridColDef } from '@mui/x-data-grid';
+
+import DataGrid from '../../atoms/data-grid';
+
+const meta: Meta<typeof DataGrid> = {
```
Comment (SGutierrez17) [inline src/stories/atoms/data-grid.stories.tsx]: Ingenieros aqui estan repitiendo Meta<typeof DataGrid> debido a que mas abajo ya lo llaman con junto al satisfies

**Review by SGutierrez17 — CHANGES_REQUESTED** [review]
Ingenieros esta funcionando correctamente el componente, ajusten el comentario que hice

File: src/stories/atoms/data-grid.stories.tsx (hunk 1 above)
Comment (paulagutierrez08) [reply src/stories/atoms/data-grid.stories.tsx]: Corregido!

**Review by SGutierrez17 — APPROVED** [review]
Good Job!

**Review by juanav22 — APPROVED** [review]

**Pull request merged**